department-of-veterans-affairs/vets-website
# MHV-60060: PHR refresh applied to all domains

## Summary
All domains are now checking for PHR updates through the PHR refresh process. 

## Related issue(s)
https://jira.devops.va.gov/browse/MHV-60060

> The following work was done for Care Summaries & Notes ([MHV-57259](https://jira.devops.va.gov/browse/MHV-57259)). This now needs to be replicated for the list pages for all other domains.
> 
> Update the PHR refresh UI changes for existing data flow
> 
> Existing user for all MR domains.  PHR refresh may be working in the background and user will have access to existing data.  Once the  PHR refresh completes:
> 
> 1- No new data is found and message displays 
> 
> 2- New data is found for current domain and message displays giving the user the ability to reload to get new updated.
> 
> See Figma for designs and details of these PHR flows:
> 
> https://www.figma.com/file/mGZRdLypKGaFsHo5xp2kaZ/Medical-Records?type=design&node-id=2776-106324&mode=design&t=2Tobqpy0T647JBJ9-0 
> 
> Feature Flag Y/N ? N
> 
> DataDog/Analytics?  Y/N N
> 
> Manual Testing Y/N ? Y-Maruf
> 
> Automated Testing Y/N ? N
> 
> Accessibility Testing Y/N ? N
> 
> UCD Validation Y/N N
> 
> Links to UCD:
> Figma Link: https://www.figma.com/file/mGZRdLypKGaFsHo5xp2kaZ/Medical-Records?type=design&node-id=2776-106324&mode=design&t=2Tobqpy0T647JBJ9-0
> Other Design Notes
> 
> Architectural Notes:
> 
> The content will always appear in this order, regardless of which is actually present (per [MHV-58869](https://jira.devops.va.gov/browse/MHV-58869)):
> 
> <h1> title
> Subtitle/text description
> New PHR refresh status component
> Print/download button
> What to know dropdown
> Pagination and Record list
> 

## Testing done
Tests were written to check that the PHR refresh process is working for each domain. 

## Screenshots
...

## What areas of the site does it impact?
MHV Medical records
- Labs and tests
- Health conditions
- Vitals
- Vaccines
- Allergies

## Acceptance criteria
- [ ] User is able to access existing data during a PHR refresh
- [ ] Message is displayed when no new data is found following FIGMA designs
- [ ] Message and reload button is displayed when new data for current domain is found following FIGMA designs
- [ ] UCD has validated changes (Amelia)
- [ ] The new PHR status content appears in the order specified in "Architectural Notes"

### Quality Assurance & Testing


### Error Handling


### Authentication
